server/server.go

```
@@ -72,9 +149,94 @@ func (n *Node) Start() {
+func (n *Node) inspectHandler(w http.ResponseWriter, r *http.Request) {
+	data := map[string]interface{}{
+		"node_id":      n.id,
+		"current_term": n.RaftNode.GetCurrentTerm(),
+		"state":        n.RaftNode.GetState().StateString(),
+		"log_entries":  n.RaftNode.GetLog(),
+		"commit_index": n.RaftNode.GetCommitIndex(),
+		"last_applied": n.RaftNode.GetLastApply(),
+		"kv_store":     n.RaftNode.KVStore.GetStore(),
+	}
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(data)
```

golangci-lint (1.62.2)

Error return value of `(*encoding/json.Encoder).Encode` is not checked

(errcheck)